src/update.c

```diff
@@ -79,7 +82,23 @@ static struct list *full_download_loop(struct list *updates, int isfailed)
 		if (file->is_deleted) {
 			continue;
 		}
-
+		/* Mix content is local, so don't queue files up for curl downloads */
+		if (file->is_mix) {
+			string_or_die(&url, "%s/%i/files/%s.tar", MIX_STATE_DIR, file->last_change, file->hash);
+			string_or_die(&filename, "%s/download/.%s.tar", state_dir, file->hash);
+			file->staging = filename;
+			ret = link(url, filename);
+			/* Try doing a regular rename if hardlink fails */
+			if (ret) {
+				if (rename(url, filename) != 0) {
```

Comment:
If the rename fails, should there a copy fallback here too?

Reply:
So now I understand the copy fallback would be the tar dance we do, and I don't think that fallback is needed here because of how it's set up. The rename fallback is enough in this case, but if we change up the locations for the local mix content maybe we can revisit this.

Reply:
Okay, sounds fine to me.